src/components/MultiHopTrade/hooks/useTradeExecution/helpers.ts

```diff
@@ -0,0 +1,43 @@
+import type { ChainId } from '@shapeshiftoss/caip'
+import type { UtxoChainAdapter } from '@shapeshiftoss/chain-adapters'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import type { AccountMetadata } from 'state/slices/portfolioSlice/portfolioSliceCommon'
+import { isUtxoChainId } from 'state/slices/portfolioSlice/utils'
+
+export type WithFromOrXpubParams = {
+  chainId: ChainId
+  accountMetadata: AccountMetadata
+  wallet: HDWallet
+}
```

Comment:
no more inputted `from` or `xpub` because these would only ever come from the wallet in our implementation anyway. A customer facing implementation capable of passing in either of those without HDWallet would not exist in this codebase :)

Reply:
🧠 
IMO the only way we'd still want `from` or `xpub` as input, is if we end up letting consumers do the signing on their side, then broadcast with us in the future but that's very very much a stretch